app/leagues/page.tsx

```diff
@@ -1,54 +1,77 @@
+import { useDataStore } from '@/store/dataStore';
+import { getGameWeek } from '@/api/apiFunctions';
 
 /**
  * Renders the leagues component.
  * @returns {JSX.Element} The rendered leagues component.
  */
 const Leagues = (): JSX.Element => {
+  const [leagues, setLeagues] = useState<ILeague[]>([]);
+  const [currentWeek, setCurrentWeek] = useState<IGameWeek['week']>(1);
+  const { user } = useDataStore((state) => state);
+
+  /**
+   * Fetches the user's leagues.
+   * @returns {Promise<void>}
+   */
+  const getLeagues = async (): Promise<void> => {
+    try {
+      const userLeagues = await getUserLeagues(user.leagues);
+      setLeagues(userLeagues);
+    } catch (error) {}
+  };
+
+  /**
+   * Fetches the current game week.
+   * @returns {Promise<void>}
+   */
+  const getCurrentGameWeek = async (): Promise<void> => {
+    try {
+      const currentWeek = await getGameWeek();
+      setCurrentWeek(currentWeek.week);
+    } catch (error) {}
+  };
+
+  useEffect(() => {
+    if (!user.id || user.id === '') {
+      return;
+    }
+
+    getLeagues();
+    getCurrentGameWeek();
+  }, [user]);
+
   return (
     <div className="Leagues mx-auto max-w-3xl pt-10">
       <h1 className="pb-10 text-center text-3xl font-bold tracking-tight">
         Your leagues
       </h1>
       <section className="grid gap-6 md:grid-cols-2">
-        <LeagueCard
-          href="/leagues"
-          leagueCardLogo="https://ryanfurrer.com/_astro/logo-dark-theme.CS8e9u7V_JfowQ.svg" // should eventually be something like league.logo
-          survivors={69}
-          title="69ers"
-          totalPlayers={138}
-        />
-        <LeagueCard
-          isEliminated={true}
-          href="/leagues"
-          survivors={69}
-          title="69ers"
-          totalPlayers={138}
-        />
-
-        <LeagueCard
-          href="/leagues"
-          survivors={11}
-          title="It's Always Sunny in Minneapolis"
-          totalPlayers={12}
-        />
-        <LeagueCard
-          href="/leagues"
-          survivors={20}
-          title="Windy City Smackdown"
-          totalPlayers={24}
-        />
-        <LeagueCard
-          href="/leagues"
-          isEliminated={true}
-          survivors={11}
-          title="Diamond Dogs"
-          totalPlayers={12}
-        />
+        {leagues.length > 0 ? (
+          leagues.map((league) => (
+            <LeagueCard
+              key={league.leagueId}
+              href={`/league/${league.leagueId}/entries/1/week/${currentWeek}`}
+              leagueCardLogo="https://ryanfurrer.com/_astro/logo-dark-theme.CS8e9u7V_JfowQ.svg" // should eventually be something like league.logo
+              survivors={league.survivors.length}
+              title={league.leagueName}
+              totalPlayers={league.participants.length}
+            />
+          ))
+        ) : (
+          <div className="text-center">
+            <p className="text-lg font-bold">Loading ...</p>
```

Comment:
nit: we need to look into Suspense instead of manually controlling this.

Reply:
@shashilo agreed.  I think we add in suspense and we can use shadcn skelleton to load a dummy group of cards.

https://ui.shadcn.com/docs/components/skeleton